src/hooks/useAssetPriceUsd.tsx

```diff
@@ -0,0 +1,31 @@
+import React, { useEffect, useMemo, useState } from 'react';
+
+// services
+import { getAssetPriceInUsd } from '../services/coingecko';
+
+const useAssetPriceUsd = (
+  chainId?: number,
+  assetAddress?: string,
+): number | null => {
+  const [assetPriceUsd, setAssetPriceUsd] = useState<number | null>(null);
+
+  useEffect(() => {
+    let shouldUpdate = true;
+
+    (async () => {
+      setAssetPriceUsd(null);
+      if (!assetAddress || !chainId) return;
+
+      const priceUsd = await getAssetPriceInUsd(chainId, assetAddress);
```

Comment:
Should we still be using the coingecko api, I'm assuming it's cos we're only pulling data for asset? Can we use the etherspot rates service for this?

Reply:
correct, but will check if I can pull native rate using etherspot SDK, thx

Reply:
@Yash-Chaubal updated the code to pull asset rates using SDK if it's not native token, also it will always falls back to coingecko if SDK fails